src/Symfony/Component/FeatureFlag/Provider/InMemoryProvider.php

```diff
@@ -30,7 +31,13 @@ public function has(string $featureName): bool
 
     public function get(string $featureName): \Closure
     {
-        return $this->features[$featureName] ?? fn() => false;
+        $feature = $this->features[$featureName] ?? fn() => false;
+
+        return function() use ($feature): mixed {
+            $arguments = $this->argumentResolver?->getArguments($feature) ?? [];
```

Comment:
Might be worth to have a InMemoryCache for argument resolver ? In case of controller they are not called multiple times. But flags might be.

Reply:
Not sure that is necessary since the feature's value is already cached by the `FeatureChecker` service.

Reply:
But if we move the argument resolver to checker it might be worth don't you think ? Or maybe I am missing somehting.